homeassistant/components/airtouch4/__init__.py

```diff
@@ -0,0 +1,93 @@
+"""The AirTouch4 integration."""
+import logging
+
+from airtouch4pyapi import AirTouch
+
+from homeassistant.components.climate import SCAN_INTERVAL
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the AirTouch4 component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up AirTouch4 from a config entry."""
+    host = entry.data[CONF_HOST]
+    airtouch = AirTouch(host)
+    try:
+        info = airtouch.GetAcs()
```

Comment:
The library doesn't support asyncio. We can't do sync work, eg I/O in the event loop. We need to use the executor thread pool or a separate thread.

Reply:
I've reworked the library to use async calls instead - hopefully this should avoid this issue now